src/components/createPostModal/index.js

```diff
@@ -1,47 +1,42 @@
-import { useState } from "react";
-import useModal from "../../hooks/useModal";
-import "./style.css";
-import Button from "../button";
-import { postPost } from "../../service/apiClient";
-import jwtDecode from "jwt-decode";
+import { useState } from "react"
+import useModal from "../../hooks/useModal"
+import "./style.css"
+import Button from "../button"
+import { postPost } from "../../service/apiClient"
+import useAuth from "../../hooks/useAuth"
 
 const CreatePostModal = ({ getAllPosts }) => {
   // Use the useModal hook to get the closeModal function so we can close the modal on user interaction
-  const { closeModal } = useModal();
+  const { closeModal } = useModal()
 
-  const [message, setMessage] = useState(null);
-  const [text, setText] = useState("");
+  const [message, setMessage] = useState(null)
+  const [text, setText] = useState("")
 
   const onChange = (e) => {
-    setText(e.target.value);
-  };
+    setText(e.target.value)
+  }
 
   const onSubmit = () => {
-    const token = localStorage.getItem("token");
-    const { userId } = jwtDecode(token);
+    const { userId } = useAuth()
```

Comment:
move this to the root of the component. Otherwise we will need to load it every time the form is submitted